analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -136,6 +144,43 @@ private bool IsEnabled(AnalyzerOptions options)
             return SupportedDiagnostics.Any(d => analyzerConfiguration.IsEnabled(d.Id));
         }
 
+        private void CheckWebConfig(SonarAnalysisContext context, CompilationAnalysisContext c)
+        {
+            foreach (var fullPath in context.GetWebConfig(c))
+            {
+                var webConfig = File.ReadAllText(fullPath);
+                if (webConfig.Contains("<system.web") && XmlHelper.ParseXDocument(webConfig) is { } doc)
+                {
+                    ReportRequestLengthViolation(doc, fullPath, c);
+                }
+            }
+        }
+
+        private void ReportRequestLengthViolation(XDocument doc, string webConfigPath, CompilationAnalysisContext c)
+        {
+            foreach (var httpRuntime in doc.XPathSelectElements("configuration/system.web/httpRuntime"))
+            {
+                if (httpRuntime.Attribute("maxRequestLength") is { } maxRequestLength
+                    && IsVulnerable(maxRequestLength.Value, MaxAllowedRequestLength)
+                    && maxRequestLength.CreateLocation(webConfigPath) is { } location)
+                {
+                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
+                }
+            }
+            foreach (var requestLimit in doc.XPathSelectElements("configuration/system.webServer/security/requestFiltering/requestLimits"))
+            {
+                if (requestLimit.Attribute("maxAllowedContentLength") is { } maxAllowedContentLength
+                    && IsVulnerable(maxAllowedContentLength.Value, MaxAllowedContentLength)
+                    && maxAllowedContentLength.CreateLocation(webConfigPath) is { } location)
+                {
+                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
+                }
+            }
+        }
+
+        private static bool IsVulnerable(string value, int limit) =>
+            !(int.TryParse(value, out var val) && val <= limit);
```

Comment:
I see `!(............ val <= limit)` and it's hurting my brain, because if `value <= limit`, it should be ok.

And then I see the `!` (this is why we have https://rules.sonarsource.com/csharp/RSPEC-1940 , but it didn't detect this)

I suggest:
```
! int.TryParse(value, out var val)
|| val > limit
```

Reply:
I've simplified it in a different way, so that it won't raise FP